projects/lib/src/configuration/config.service.ts

```diff
@@ -29,57 +18,131 @@ export interface ContainerConfig extends ComponentConfig {
+  }),
+
+  on(addConfig, (state, {config}) => {
+    if(state.config[config.id]) {
+      throw new Error(`Config ${config.id} already exists.`);
+    }
+    state.config[config.id] = config;
+    return state;
+  }),
+
+  on(removeConfig, (state, {id}) => {
+    delete state.config[id];
+    return state;
+  }),
+
+  on(updateConfig, (state, {config}) => {
+    if(!Array.isArray(config)) config = [config];
+    for(let c of config) {
+      state.config[c.id] = c;
+    }
+    return state;
+  }),
+
+  on(updateNoHistoryConfig, (state, {config}) => {
+    if(!Array.isArray(config)) config = [config];
+    for(let c of config) {
+      state.config[c.id] = c;
+    }
+    return state;
+  })
+);
+
+const {
+  selectCanUndo,
+  selectCanRedo,
+} = createHistorySelectors<State, State>(selectConfig);
+
+
 @Injectable({ providedIn: 'root' })
 export class ConfigService {
-  store: Store;
-  historyState: any;
 
-  init(initialValue: ComponentConfig[]) {
-    const { state, config } = createState(
-      withEntities<ComponentConfig>({ initialValue })
-    );
-    this.store = new Store({ name: 'config', state, config });
-    this.historyState = stateHistory(this.store, { maxAge: Infinity });
-    this.store.subscribe(console.log);
+  constructor(public store: Store<State>){
+    this.store.subscribe(v => console.log(v));
   }
 
-  set(config: ComponentConfig[]) {
-    this.store.update(
-      deleteAllEntities(),
-      addEntities(config)
-    );
+  public init(config: ComponentConfig[]) {
+    this.store.dispatch(initConfig({config}));
+  }
+
+  public set(config: ComponentConfig[]) {
+    this.store.dispatch(setConfig({config}));
   }
 
   public watchAllConfig(): Observable<ComponentConfig[]> {
-    return this.store.pipe(selectAllEntities());
+    return this.store.select(selectAll);
   }
 
   public getAllConfig(): ComponentConfig[] {
-    return this.store.query(getAllEntities());
+    let config: ComponentConfig[] = [];
+    this.store.select(selectAll).pipe(take(1)).subscribe(c => config = c);
+    return config;
   }
 
   public watchConfig(id: string): Observable<ComponentConfig> {
-    //console.log('watch:', id);
     this.getConfig(id); // Ensure a value exists (if 'id' has no config)
-    return this.store.pipe(
-      selectEntity(id),
+    const selector = createSelector(selectConfig, s => s.config[id]);
+    return this.store.select(selector).pipe(
       filter(config => config !== undefined),
-      //tap((config) => console.log('change:', config)),
-      map((config) => JSON.parse(JSON.stringify(config)))
+      map((config: ComponentConfig) => JSON.parse(JSON.stringify(config)))
     );
   }
 
   private _getConfig(id: string): ComponentConfig | undefined {
-    return this.store.query(getEntity(id));
+    let config: ComponentConfig | undefined;
+    const selector = createSelector(selectConfig, s => s.config[id]);
+    this.store.select(selector)
+      .pipe(take(1))
+      .subscribe(c => config = c);
+    return config;
```

Comment:
Here "config" is returned before `subscribe()` was done
`subscribe()` is asynchronous

to work you should:
```javascript
async private _getConfig(...) {
  ...
  await this.store.select(...)
  return config;
}
```